### ISSUE TITLE
Int16 input argument

### ISSUE THREAD

**Issue by fcontijoch** [tomdoel/pulmonarytoolkit#13]
Hello.

This software really looks great. However, when we try to run a variety of different functions including segmentation of the lungs, lobes, and segments, I am receiving the following error:

The plugin PTKLeftAndRightLungs failed with the following error: Underfined function ‘PTKWatershedFromStartingPoints’ for input arguments type ‘int16’;

The airway segmentation works fine.

Thanks

**Comment by tomdoel** [comment]
Hi,

It sounds like the C++ files have not compiled correctly. Normally these are compiled automatically the first time you start the application, but if a compatible compiler is not available then this will not work.

If the files are compiled they will appear in the `pulmonarytoolkit\bin` folder. If you don't see a file `PTKWatershedFromStartingPoints.<something>` (where `<something>` depends on your platform) then the mex file has not yet been compiled. Use the following commands to check you have a compiler installed and then force recompilation.

You can check if you have a compiler installed by running `mex -setup` on the Matlab Command Window. It should say something like `MEX configured to use .... for C language compilation`. If no compiler is found, you'll need to install one - see here for the [current list of Matlab supported compilers](https://uk.mathworks.com/support/compilers/R2016b/).

Once you have a compiler installed, you can force PTK to recompile the mex files using the following commands in the Matlab Command Window:
```
ptk_main = PTKMain();
ptk_main.Recompile();
```

If you get a failure when compiling, it will likely be due to a problem with the compiler or compiler settings.

**Comment by fcontijoch** [comment]
Thanks. It looks like the files were compiling correctly. However, the bin folder had not been added to the path. I added it manually and now running ptk does not give that error.

Thanks!

**Comment by tomdoel** [comment]
Found the problem - thanks for alerting me to this. Normally the app sets up the paths correctly, but the first time it's run 'bin' doesn't exist and so doesn't get added to the path. (Previously, the project was stored under subversion and an empty 'bin' folder was created on checkout, but git doesn't do this.) I will put in a fix in the next release.

In case anyone experiences the same problem, the paths can be set up again using the following command:
```
PTKAddPaths force
```

**Comment by tomdoel** [comment]
For a test dataset, you can try the "CARCINOMIX" dataset from the [Osirix DICOM Image Library](http://www.osirix-viewer.com/resources/dicom-image-library/) - that works reasonably well with PTK for airway, lung and lobe segmentation.

**Comment by tomdoel** [comment]
Fixed in v0.7
